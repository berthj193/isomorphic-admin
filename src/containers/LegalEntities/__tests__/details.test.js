import React from 'react';
import LegalEntityDetails, { LegalEntityDetails as RawLegalEntityDetails } from '../details';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('<LegalEntityDetails />', () => {
  const defaultParams = {
    match: {
      params: {},
    },
    legal_entities: [{
      id: 1,
      name: 'foo',
      vat_number: '123',
    }],
  };
  const getComponent = (props = {}) => shallow(<RawLegalEntityDetails {...defaultParams} {...props} />);
  const mockStore = configureStore();

  it('should render loading page', () => {
    expect(toJson(getComponent({ legal_entities: [{}] }))).toMatchSnapshot();
  });

  it('renderList should return list of elements', () => {
    expect(toJson(getComponent().instance().renderList(['foo', 'bar'])())).toMatchSnapshot();
  });

  it('loadDetailList should return list of details to render', () => {
    expect(getComponent().instance().loadDetailList().length).toBeGreaterThan(1);
  });

  it('should render details with provider', () => {
    expect(toJson(shallow(<LegalEntityDetails store={mockStore()} />))).toMatchSnapshot();
  });
});
