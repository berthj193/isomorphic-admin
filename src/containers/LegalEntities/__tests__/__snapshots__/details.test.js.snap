// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LegalEntityDetails /> renderList should return list of elements 1`] = `undefined`;

exports[`<LegalEntityDetails /> should render details with provider 1`] = `
<ComponentWithProvider
  startRequest={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`<LegalEntityDetails /> should render loading page 1`] = `
<PageLoading
  style={
    Object {
      "height": "50vh",
    }
  }
/>
`;
